Doppler.Sap/Services/BillingServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,37 @@
+using Doppler.Sap.Models;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Doppler.Sap.Services
+{
+    public static class ServicesCollectionExtensions
+    {
+        public static void AddServices(this IServiceCollection services, IConfiguration configuration)
+        {
+            services.AddSingleton<DummyBillingService>();
+            services.AddTransient<BillingService>();
+            services.AddSingleton<DummyBusinessPartnerService>();
+            services.AddTransient<BusinessPartnerService>();
+
+            var section = configuration.GetSection("SapConfig");
+            var sapSettings = new SapConfig();
+            section.Bind(sapSettings);
+
+            services.AddSingleton<IBillingService>(provider =>
+            {
+                if (!section.Exists() || sapSettings.UseDummyData)
+                    return provider.GetRequiredService<DummyBillingService>();
+
+                return provider.GetRequiredService<BillingService>();
+            });
+
+            services.AddSingleton<IBusinessPartnerService>(provider =>
+            {
+                if (!section.Exists() || sapSettings.UseDummyData)
+                    return provider.GetRequiredService<DummyBusinessPartnerService>();
+
+                return provider.GetRequiredService<BusinessPartnerService>();
+            });
```

Comment:
Please take into account my comments related to `IBillingService` for this one too.

Reply:
Ok, done